controllers/machine_controller_phases.go

```diff
@@ -226,6 +231,10 @@ func (r *MachineReconciler) reconcileInfrastructure(ctx context.Context, cluster
 	}
 	infraConfig := infraReconcileResult.Result
 
+	// Report a summary of current status of the infrastructure object defined for this machine.
+	infrastructureReadyCondition := conditions.Mirror(conditions.UnstructuredGetter(infraConfig), clusterv1.InfrastructureReadyCondition)
+	conditions.Set(m, infrastructureReadyCondition)
```

Comment:
Should this also fall back to using `external.IsReady(infraConfig)` to support intrastructure providers that haven't adopted conditions yet?

Reply:
Implemented fall-back